packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -215,6 +222,24 @@ export default function<T, P, I, TI, HI, PI, C, CC, CX, PL>(
     return workInProgress.child;
   }
 
+  function updateProfiler(current, workInProgress) {
+    const nextProps = workInProgress.pendingProps;
+    if (enableProfileModeMetrics) {
+      // Start render timer here and push start time onto queue
+      markActualRenderTimeStarted(workInProgress);
+
+      // Let the "complete" phase know to stop the timer,
+      // And the scheduler to record the measured time.
+      workInProgress.effectTag |= Update;
+    } else if (workInProgress.memoizedProps === nextProps) {
```

Comment:
If enableProfileModeMetrics is enabled, we should still bailout on equality, right? Seems odd to disable that optimization.

Reply:
Yes, good catch. This should have been two ifs rather than an if/else